patterns-ai-core/langchainrb
# Add fallback tool_call_id to Ollama tool calls

### Description
Currently, the response for Ollama tool calls lack a `tool_call_id`. This is necessary to support output from multiple tools, as seen in the templates of llama3.1 and mistral-nemo. To address this, we should use the responded `function_name` as a fallback for `tool_call_id`.

For a permanent fix for Ollama, please refer to my GitHub issue for Ollama: [#6462](https://github.com/ollama/ollama/issues/6462). My implementation is prepared for future adjustments by Ollama.

### Changes made
- Added fallback for `tool_call_id` for Ollama tool calls
- Used the responded `function_name` as the `tool_call_id`

### Why this change is needed
This change is essential to ensure that outputs from multiple tools can be properly identified and managed.

### How to test
1. Make a tool call using a supported Ollama model (llama3.1, ...).
2. Verify that the `tool_call_id` is present and matches the `function_name`.
3. Ensure that multiple tool calls can be distinguished by their `tool_call_id`.

### Checklist

Thank you for your review!
